electrum/lnpeer.py

```diff
@@ -1720,6 +1725,11 @@ def on_channel_ready(self, chan: Channel, payload):
         self.logger.info(f"on_channel_ready. channel: {chan.channel_id.hex()}")
         if chan.peer_state != PeerState.GOOD:  # should never happen
             raise Exception(f"received channel_ready in unexpected {chan.peer_state=!r}")
```

Comment:
This test that chan.peer_state is GOOD has been removed from various places (see below). Why keep it in `on_channel_ready`?

Reply:
ah, I see, you are not calling `can_update_ctx` in that specific case, I guess that's why